queue/queue_private.js

```diff
@@ -25,8 +35,13 @@ queueApp.controller("QueuePrivateController", function($scope, $controller, Queu
 	};
 
 	$scope.completeAppointment = function() {
+		if (this.selectedVolunteer == null) {
```

Comment:
@CollinsSpencer For some reason, I was unable to do $scope.selectedVolunteer and had to do this.selectedVolunteer (note that selectedVolunteer is the ng-model on the select). Any ideas why? Idk if it's frowned upon to use this or anything in this Angular stuff.

Reply:
You need ```$scope.client.selectedVolunteer```

Reply:
Yeah. In the private.php you need ```ng-model="client.selectedVolunteer"```. Currently you're creating a selectedVolunteer variable that is used for all of the clients. Then you can use ```$scope.client.selectedVolunteer``` on this line and ```$scope.client.selectedVolunteer.userId``` in the call to QueueService.finishAppointment().